src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,421 @@
+        /// <param name="dutyCyclePercentage">The duty cycle as a fraction.</param>
+        /// <returns></returns>
+        public virtual PwmChannel CreatePwmChannel(
+            int chip,
+            int channel,
+            int frequency = 400,
+            double dutyCyclePercentage = 0.5)
+        {
+            return new ArduinoPwmChannel(this, chip, channel, frequency, dutyCyclePercentage);
+        }
+
+        /// <summary>
+        /// Creates an anlog controller for this board.
+        /// </summary>
+        /// <param name="chip">Must be 0</param>
+        /// <returns>An <see cref="AnalogController"/> instance</returns>
+        public virtual AnalogController CreateAnalogController(int chip)
+        {
+            return new ArduinoAnalogController(this, SupportedPinConfigurations, PinNumberingScheme.Logical);
+        }
+
+        /// <summary>
+        /// Special function to read DHT sensor, if supported
+        /// </summary>
+        /// <param name="pinNumber">Pin Number</param>
+        /// <param name="dhtType">Type of DHT Sensor: 11 = DHT11, 22 = DHT22, etc.</param>
+        /// <param name="temperature">Temperature</param>
+        /// <param name="humidity">Relative humidity</param>
+        /// <returns>True on success, false otherwise</returns>
+        public bool TryReadDht(int pinNumber, int dhtType, out Temperature temperature, out RelativeHumidity humidity)
+        {
+            if (!_supportedPinConfigurations[pinNumber].PinModes.Contains(SupportedMode.DHT))
+            {
+                temperature = default;
+                humidity = default;
+                return false;
+            }
+
+            return Firmata.TryReadDht(pinNumber, dhtType, out temperature, out humidity);
+        }
+
+        /// <summary>
+        /// Standard dispose pattern
+        /// </summary>
+        protected virtual void Dispose(bool disposing)
+        {
+            // Do this first, to force any blocking read operations to end
+            if (_dataStream != null)
+            {
+                _dataStream.Close();
+                _dataStream.Dispose();
+                _dataStream = null!;
+            }
+
+            if (_serialPort != null)
+            {
+                _serialPort.Dispose();
+                _serialPort = null;
+            }
+
+            if (_firmata != null)
+            {
+                _firmata.OnError -= FirmataOnError;
+                _firmata.Close();
+                _firmata.Dispose();
+            }
+
+            _firmata = null;
+        }
+
+        /// <summary>
+        /// Dispose this instance
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        internal void EnableSpi()
+        {
+            _spiEnabled++;
+            if (_spiEnabled == 1)
+            {
+                _firmata?.EnableSpi();
```

Comment:
can you always enable it? Also might make sense to check for negative ref count

Reply:
No, since that blocks the pins for other uses (if you don't need SPI, you can use the pins for GPIO).